### PR TITLE
Object manipulation 3

### PR DESCRIPTION
**Pull request by vkahlon**
Demo:

https://user-images.githubusercontent.com/47346471/149247834-234cd308-da2e-432f-b0c3-1bf97b6c4465.mp4

### CODE REVIEW

File: object-manipulation-3/main.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+var heartsDeck = createDeck('hearts', heart);
+var diamond = [];
+var diamondsDeck = createDeck('diamonds', diamond);
+var the26Two = heartsDeck.concat(diamondsDeck);
+var fullDeck = the26One.concat(the26Two);
+var shuffledCards = _.shuffle(fullDeck);
+var hand1 = [shuffledCards[0], shuffledCards[1]];
+var hand2 = [shuffledCards[2], shuffledCards[3]];
+var hand3 = [shuffledCards[4], shuffledCards[5]];
+var hand4 = [shuffledCards[6], shuffledCards[7]];
```
Comment (codingcodymiller) [inline object-manipulation-3/main.js:17]: When you deal a card to a player, it should be removed from the deck, just like real cards.

File: object-manipulation-3/main.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      return card;
+    }
+  }
+  return theTable;
+}
+var grabWinnerList = _.sortBy(ranks, ['points']);
+console.log('The winner is ' + grabWinnerList[3].name + ' who had ' + grabWinnerList[3].points + ' points.');
+console.log('Second place is ' + grabWinnerList[2].name + ' who had ' + grabWinnerList[2].points + ' points.');
+console.log('Third place is ' + grabWinnerList[1].name + ' who had ' + grabWinnerList[1].points + ' points.');
+console.log('Fourth place is ' + grabWinnerList[0].name + ' who had ' + grabWinnerList[0].points + ' points.');
```
Comment (codingcodymiller) [inline object-manipulation-3/main.js:94]: What if there's a tie? :thinking:

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by vkahlon** [comment]
tie

File: object-manipulation-3/main.js [hunk 3]
```diff
@@ -1,94 +1,95 @@
...
+  if (grabWinnerList[getPlayerCount].points === grabWinnerList[getPlayerCount - 1].points) {
+    console.log('TIE BREAKER!');
+    var newGame = [];
+    var grabTieBreakerPlayer1 = { name: grabWinnerList[2].name, hand: null };
+    var grabTieBreakerPlayer2 = { name: grabWinnerList[3].name, hand: null };
+    newGame.push(grabTieBreakerPlayer1);
+    newGame.push(grabTieBreakerPlayer2);
+    var startNewRound = startsGame(newGame);
+    console.log('The winner is ' + startNewRound[1].name + ' who had ' + startNewRound[1].points + ' points.');
+    console.log('The runner-up is ' + startNewRound[0].name + ' who had ' + startNewRound[0].points + ' points.');
```
Comment (codingcodymiller) [inline object-manipulation-3/main.js:89]: What if **_this_** game has a tie? 🤔

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by vkahlon** [comment]
Screen Shot 2022-01-13 at 5 16 48 PM

File: object-manipulation-3/main.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    }
+    var newJack = { rank: 'Jack', suit: suit };
+    list.push(newJack);
+    var newQueen = { rank: 'Queen', suit: suit };
+    list.push(newQueen);
+    var newKing = { rank: 'King', suit: suit };
+    list.push(newKing);
+    var newAce = { rank: 'Ace', suit: suit };
+    list.push(newAce);
+    return list;
```
Comment (thebearingedge) [inline object-manipulation-3/main.js:45]: If you use two nested loops you can create every card the same way inside of the inner loop.

```js
var ranks = ['ace', 'king', 'queen', 'jack', 10, 9, 8, 7, 5, 4, 3, 2];
var suits = ['hearts', 'clubs', 'diamonds', 'spades'];
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**